### PR TITLE
[OSD-13229] Include limited-support metric in osd-metrics-exporter

### PR DESCRIPTION
**Pull request by aliceh**
In this commit, I added a gauge to include the limited-support metric in osd-metrics-exporter.

### CODE REVIEW

**Comment by T0MASD** [comment]
It would be great to see unit tests as part of this MR, for reference an example proxy metrics test https://github.com/openshift/osd-metrics-exporter/blob/main/pkg/controller/proxy/proxy_controller_test.go

File: pkg/controller/limited-support/limited_support_controller.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+func (r *ReconcileLimitedSupportConfigMap) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Limited Support ConfigMap")
+
+	// Fetch the ConfigMap openshift-osd-metrics/limited-support
+	cfgMap := &corev1.ConfigMap{}
+	ns := limitedSupportConfigMapNamespace
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: limitedSupportConfigMapName}, cfgMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (fahlmant) [inline pkg/controller/limited-support/limited_support_controller.go:96]: Should this set the metric to false?

Comment (aliceh) [reply pkg/controller/limited-support/limited_support_controller.go:96]: Good catch. I will change it. I accidentally left it in when I copied it from the configmap_controller

**Comment by aliceh** [comment]
@sam-nguyen7  I added the unit tests

File: pkg/controller/limited-support/limited_support_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
+package limited_support
```
Comment (sam-nguyen7) [inline pkg/controller/limited-support/limited_support_test.go:1]: Can we change the file name to `limited_support_controller_test.go` to match the pattern of this repo?

Comment (aliceh) [reply pkg/controller/limited-support/limited_support_test.go]: Absolutely. Good catch!

**Pull request merged**